conans/model/build_info.py

```diff
@@ -12,6 +12,18 @@
 DEFAULT_FRAMEWORK = "Frameworks"
 
 
+class DefaultOrderedDict(OrderedDict):
```

Comment:
For me the problem with this inheritance is that the users will be able to do more crazy things:

```python
def package_info(self):
        self.cpp_info.components["mycomp"] = "Something"  # Will fail later with weird backtraces
        for name, comp in self.cpp_info.components.items():
                comp.includedirs = "some header"
```

Not saying this is bad per-se, maybe it is a good and useful feature, not very problematic, but as we know, if users can  do it, they will do it, so better be sure there are no rough edges there.

Reply:
I initially went for the composition pattern you suggested, but then there were errors that the object was not serializable for the JSON output, I understand what you say there, but I tried not to over-protect the user and assume they are going to use it properly